lib/state-controller.js

```diff
@@ -38,140 +38,57 @@ var StateController = {
 
   downloadedStream: null,
 
-  isKiteSupported: function() {
-    return new Promise((resolve, reject) => {
-      setTimeout(() => {
-        os.platform() === 'darwin' && os.release() >= '10.10.0' ?
-          resolve() :
-          reject({ type: 'bad_state', data: this.STATES.UNSUPPORTED });
-      }, 0);
-    });
+  isKiteSupported () {
+    return os.platform() === 'darwin' && os.release() >= '10.10.0'
+      ? Promise.resolve()
+      : Promise.reject({ type: 'bad_state', data: this.STATES.UNSUPPORTED });
   },
 
-  isKiteInstalled: function() {
-    return new Promise((resolve, reject) => {
-      this.isKiteSupported().then(() => {
-        var ls = child_process.spawnSync('ls', [this.KITE_APP_PATH.installed]);
-        ls.stdout.length !== 0 ?
-          resolve() :
-          reject({ type: 'bad_state', data: this.STATES.UNINSTALLED });
-      }, (err) => {
-        reject(err);
-      });
-    });
+  isKiteInstalled () {
+    return this.isKiteSupported().then(() => {
+      if (!fs.existsSync(this.KITE_APP_PATH.installed)) {
+        throw { type: 'bad_state', data: this.STATES.UNINSTALLED }
+      }
+    })
```

Comment:
Initially I would have advocated for [standard](http://standardjs.com/) as it was the styles used through Atom. But it appears they have been convinced by FB to switch to eslint styles (with semi-colons and such).

Personnally I don't see any value in keeping semi-colons around, it's not C or PHP but JavaScript. But given that semi-colons are already everywhere in the code I'll stick with that. 

> maybe we should add a linter to the CI process

Yup I'm always doing that in my project.

> could always do that in a separate branch too. open to suggestions

I'll take a look at an eslint config so that we don't need to create another branch just for that.